src/rviz/default_plugin/markers/shape_marker.cpp

```diff
@@ -79,9 +79,12 @@ void ShapeMarker::onNewMessage( const MarkerConstPtr& old_message,
     handler_->addTrackedObjects( shape_->getRootNode() );
   }
 
-  Ogre::Vector3 pos, scale, scale_correct;
+  Ogre::Vector3 pos(0,0,0);
+  Ogre::Vector3 scale(0,0,0);
+  Ogre::Vector3 scale_correct(0,0,0);
   Ogre::Quaternion orient;
-  transform(new_message, pos, orient, scale);
+
+  if(!transform(new_message, pos, orient, scale)) { return; }
```

Comment:
Wouldn't it suffice to have this line fixed, i.e. just return on failure?
All the uninitialized variables wouldn't be used in this case.